src/NuGetForUnity/Editor/NugetPackageInstaller.cs

```diff
@@ -196,36 +198,125 @@ private static bool Install([NotNull] INugetPackage package, bool refreshAssets,
+                            if (bestFrameworkMatch.Value != null)
+                            {
+                                NugetLogger.LogVerbose(
+                                    "Selecting target framework directory '{0}' and language 'cs' as best match for the package {1}",
+                                    bestFrameworkMatch.Key,
+                                    package);
+
+                                PackageContentManager.ExtractPackageSources(bestFrameworkMatch.Value, baseDirectory);
+                            }
+                            else if (csFiles.TryGetValue("any", out var anyFrameworks))
+                            {
+                                NugetLogger.LogVerbose(
+                                    "Selecting target framework directory 'any' and language 'any' as best match for the package {1}",
+                                    package);
+
+                                PackageContentManager.ExtractPackageSources(anyFrameworks, baseDirectory);
+                            }
+                            else
+                            {
+                                Debug.LogWarningFormat(
+                                    "Couldn't find a source code folder with a supported target-framework for the package {0}",
+                                    package);
+                            }
+                        }
+                        else if (anyFiles.Count > 0)
+                        {
+                            bestFrameworkMatch = TargetFrameworkResolver.TryGetBestTargetFramework(anyFiles, framework => framework.Key);
+                            if (bestFrameworkMatch.Value != null)
+                            {
+                                NugetLogger.LogVerbose(
+                                    "Selecting target framework directory '{0}' and language 'any' as best match for the package {1}",
+                                    bestFrameworkMatch.Key,
+                                    package);
+
+                                PackageContentManager.ExtractPackageSources(bestFrameworkMatch.Value, baseDirectory);
+                            }
+                            else if (anyFiles.TryGetValue("any", out var anyFrameworks))
+                            {
+                                NugetLogger.LogVerbose(
+                                    "Selecting target framework directory 'any' and language 'any' as best match for the package {1}", package);
+
+                                PackageContentManager.ExtractPackageSources(anyFrameworks, baseDirectory);
+                            }
+                            else
+                            {
+                                Debug.LogWarningFormat(
+                                    "Couldn't find a source code folder with a supported target-framework for the package {0}",
+                                    package);
+                            }
```

Comment:
You can simplify this using something like:
```cs
if (csFiles.Count > 0)
{
	TryExtractBestFrameworkSources(csFiles, 'cs');
}
else if (anyFiles.Count > 0)
{
	TryExtractBestFrameworkSources(anyFiles, 'any');
}

private void TryExtractBestFrameworkSources([NotNull] [ItemNotNull] IReadOnlyCollection<KeyValuePair<string, List<ZipArchiveEntry>> frameworks, [NotNull] string sourceDirName)
{
	var bestFrameworkMatch = TargetFrameworkResolver.TryGetBestTargetFramework(frameworks, framework => framework.Key);
	var frameworkKey = bestFrameworkMatch.Key ?? "any";
	if (frameworks.TryGetValue(frameworkKey, out var bestFramework))
	{
		NugetLogger.LogVerbose(
			"Selecting target framework directory '{0}' and language '{1}' as best match for the package {2}",
			bestFramework,
			sourceDirName,
			package);

		PackageContentManager.ExtractPackageSources(bestFramework, baseDirectory);
	}
	else
	{
		Debug.LogWarningFormat(
			"Couldn't find a source code folder with a supported target-framework for the package {0}",
			package);
	}
}
```

Reply:
yes without the "extraction of a function" it is definitely to much copy-paste.

Reply:
Simplified but without `[ItemNotNull]` as Rider said `Applying nullness annotation to value type is meaningless`.